app/plugin_requests.go

```diff
@@ -74,6 +74,8 @@ func (a *App) ServeInterPluginRequest(w http.ResponseWriter, r *http.Request, so
 		SourcePluginId: sourcePluginId,
 	}
 
+	r.Header.Set("Mattermost-Plugin-ID", sourcePluginId)
```

Comment:
I am not sure if we should follow the CanonicalHeaderKey pattern here, i.e:
`r.Header.Set("Mattermost-Plugin-Id", sourcePluginId)`